cranelift/codegen/build.rs

```diff
@@ -14,10 +14,31 @@
 // The build script expects to be run from the directory where this build.rs file lives. The
 // current directory is used to find the sources.
 
+// We use the deprecated SipHasher from std::hash in order to verify
+// that ISLE sources haven't changed since the generated source was
+// last regenerated.
+//
+// We use this instead of a stronger and more usual content hash, like
+// SHA-{160,256,512}, because it's built into the standard library and
+// we don't want to pull in a separate crate. We try to keep Cranelift
+// crate dependencies as intentionally small as possible. In fact, we
+// used to use the `sha2` crate for SHA-512 and this turns out to be
+// undesirable for downstream consumers (see #3609).
+//
+// Why not the recommended replacement
+// `std::collections::hash_map::DefaultHasher`? Because we need the
+// hash to be deterministic, both between runs (i.e., not seeded with
+// random state) and across Rust versions.
+//
+// If `SipHasher` is ever actually removed from `std`, we'll need to
+// find a new option, either a very small crate or something else
+// that's built-in.
+#![allow(deprecated)]
```

Comment:
Could this be moved to the `compute_manifest` function to localize the allow-deprecated-things to a smaller scope? (moving the import of `SipHasher` down there too)

Reply:
Done!